openquake/risklib/asset.py

```diff
@@ -577,7 +577,8 @@ def build_asset_array(assets_by_site, tagnames=(), time_event=None):
     int_fields = [(str(name), U32) for name in tagnames]
     tagi = {str(name): i for i, name in enumerate(tagnames)}
     asset_dt = numpy.dtype(
-        [('id', '<S20'), ('ordinal', U32), ('lon', F32), ('lat', F32),
+        [('id', (numpy.string_, valid.ASSET_ID_LENGTH)),
```

Comment:
I recall that you had to change it to 20 chars in the first place because of an hdf5 bug (#5756). Is it sufficient to have a fixed-length instead of a variable-length string, and the number of characters doesn't matter?